File: data/multilang_comparison/program_cpp_vector/src/func51.c
#include "program_cpp_vector.h" 
array_t* func51(array_t_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   array_t* array38;
   if (pCounter > 0) {
      array38 = vars->data[--pCounter];
      array38->refC++;
      DEBUG_COPY(array38->id);
   } else {
      array38 = (array_t*)malloc(sizeof(array_t));
      array38->size = 683;
      array38->refC = 1;
      array38->id = 38;
      array38->data = (unsigned int*)malloc(array38->size*sizeof(unsigned int));
      memset(array38->data, 0, array38->size*sizeof(unsigned int));
      DEBUG_NEW(array38->id);
   }
   unsigned int loop13 = 0;
   unsigned int loopLimit13 = (rand()%loopsFactor)/4 + 1;
   for(; loop13 < loopLimit13; loop13++) {
      for (int i = 0; i < array38->size; i++) {
         array38->data[i]++;
      }
   }
   for (int i = 0; i < array38->size; i++) {
      if (array38->data[i] == 28) { 
         return array38;
      }
   }
   return array38;
}

File: data/multilang_comparison/program_go/program_go/src/func51.go
package main
import "math/rand"
var _ = rand.Intn
func Func51(vars *ArrayParam, loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   var array38 *Array
   if pCounter > 0 {
       array38 = vars.Data[pCounter-1]
       pCounter--
       array38.RefC++
   } else {
       array38 = &Array{
           Size: 683,
           RefC: 1,
           Id: 38,
           Data: make([]uint32, 683),
       }
   }
   var loop13 int = 0;
   var loopLimit13 int = (50)/4 + 1;
   for ; loop13 < loopLimit13; loop13++ {
      for i := 0; i < array38.Size; i++ {
          array38.Data[i]++
      }
   }
   for i := 0; i < array38.Size; i++ {
       if array38.Data[i] == 28 {
           return array38
       }
   }
   return array38;
}

